File: part-1/set-2/python/challenge-1/employee_test.py
import unittest

from employee import Employee

class EmployeeTest(unittest.TestCase):

    def testClassInstantiation(self):
        emp = Employee()
        # Add assertions to test the class instantiation
        self.assertEqual("", emp.getName())
        self.assertEqual(0, emp.getAge())
        self.assertEqual("", emp.getDepartment())

    def testConstructorOrder(self):
        # Test constructor with name, age, department order
        emp1 = Employee("John Doe", 30, "Sales")

        # Add assertions to test the constructor with name, age, department order
        self.assertEqual("John Doe", emp1.getName())
        self.assertEqual(30, emp1.getAge())
        self.assertEqual("Sales", emp1.getDepartment())

        # Test constructor with department, name, age order
        emp2 = Employee("Marketing", "Alice", 25) # FAILS # Due to incorrect arg order.
        ## In Python, using keyword arguments lets us pass args to a function
        # or method in a different order than they are defined.
        #emp2 = Employee(department="Marketing", name="Alice", age=25) # PASSES #

        # Add assertions to test the constructor with department, name, age order
        self.assertEqual("Alice", emp2.getName())
        self.assertEqual(25, emp2.getAge())
        self.assertEqual("Marketing", emp2.getDepartment())

    def testGettersAndSetters(self):
        emp = Employee()

        emp.setName("Alice")
        # Add assertions to test the setName() method
        self.assertEqual("Alice", emp.getName())

        emp.setAge(25)
        # Add assertions to test the setAge() method
        self.assertEqual(25, emp.getAge())

        emp.setDepartment("Marketing")
        # Add assertions to test the setDepartment() method
        self.assertEqual("Marketing", emp.getDepartment())

if __name__ == '__main__':
    unittest.main()

File: part-1/set-2/java/challenge-1/EmployeeTest.java
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;
public class EmployeeTest {

    @Test
    public void testClassInstantiation() {
        Employee emp = new Employee();
        // Add assertions to test the class instantiation
        Assertions.assertEquals("", emp.getName());
        Assertions.assertEquals(0, emp.getAge());
        Assertions.assertEquals("", emp.getDepartment());
    }

    @Test
    public void testConstructorOrder() {
        // Test constructor with name, age, department order
        Employee emp1 = new Employee("John Doe", 30, "Sales");

        // Add assertions to test the constructor with name, age, department order
        Assertions.assertEquals("John Doe", emp1.getName());
        Assertions.assertEquals(30, emp1.getAge());
        Assertions.assertEquals("Sales", emp1.getDepartment());

        // Test constructor with department, name, age order
        Employee emp2 = new Employee("Marketing", "Alice", 25);

        // Add assertions to test the constructor with department, name, age order
        Assertions.assertEquals("Alice", emp2.getName());
        Assertions.assertEquals(25, emp2.getAge());
        Assertions.assertEquals("Marketing", emp2.getDepartment());
    }

    @Test
    public void testGettersAndSetters() {
        Employee emp = new Employee();

        emp.setName("Alice");
        // Add assertions to test the setName() method
        Assertions.assertEquals("Alice", emp.getName());

        emp.setAge(25);
        // Add assertions to test the setAge() method
        Assertions.assertEquals(25, emp.getAge());

        emp.setDepartment("Marketing");
        // Add assertions to test the setDepartment() method
        Assertions.assertEquals("Marketing", emp.getDepartment());
    }
}
